qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -919,33 +865,48 @@ def _run_analysis(
             curve_fitter = self._get_option("curve_fitter")
             formatted_data = self._data(label="fit_ready")
 
-            # Generate fit options
-            fit_candidates = self._setup_fitting(**extra_options)
+            # Generate algorithmic initial guess
+            fit_opt_dicts = self._generate_fit_guesses()
+            if isinstance(fit_opt_dicts, dict):
+                fit_opt_dicts = [fit_opt_dicts]
+
+            # Format parameters
+            fit_candidates = set()
+            for fit_opt_dict in fit_opt_dicts:
+                fit_candidate = FitOptions(self._fit_params())
```

Comment:
This is difficult because the key of `guess` and `defaults` conflicts (both contain `p0` and `bounds`).

Reply:
Your code assumes anyhow that if the user provides the `defaults` then these `defaults` are used instead of the `guess` from the algorithmically generated initial guesses. I.e. in `FitOptions` you could do
```python
def __init__(self, p0: Dict[str, float] = None, bounds: Dict[str, float] = None, user_defaults: Dict[str, Any]):
    self.p0 = p0
    self.bounds = bounds
    
    # Override any given arguments with those from the user
    self.p0 = user_defaults.get("p0", None)
    self.bounds = user_defaults.get("bounds", None)
```
or alternatively
```python
user = {key+"_user": self._get_option(key) for key in ["p0", "bounds"]  # in curve_analysis

# In FitOptions and omitting the type hints (all are Dict[str, float])
def __init__(self, p0=None, bounds=None, p0_user=None, bounds_user=None):
    self.p0 = p0
    self.bounds = bounds
    
    # override algorithmic defaults with user provided ones
    self.p0 = p0_user
    self.bounds = bounds_user
```
Would that work?